### PR TITLE
Remove the option to support JDK7 from Java generator and templates

### PR DESCRIPTION
**Pull request by wing328**
Default all Java client, server generators to support JDK8 at the minimum

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (5.3.0), `6.0.x`
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

cc @bbdouglas (2017/07) @sreeshas (2017/08) @jfiala (2017/08) @lukoyanov (2017/09) @cbornet (2017/09) @jeff9finger (2018/01) @karismann (2019/03) @Zomzog (2019/04) @lwlee2608 (2019/10)

### CODE REVIEW

**Comment by wing328** [comment]
Appveyor failure not related to this change.

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractJavaCodegen.java [hunk 1]
```diff
@@ -933,14 +907,9 @@ public String toDefaultValue(Schema schema) {
...
 
-            String typeDeclaration = String.format(Locale.ROOT, "String, %s", getTypeDeclaration(getAdditionalProperties(schema)));
-            Object java8obj = additionalProperties.get("java8");
-            if (java8obj != null) {
-                Boolean java8 = Boolean.valueOf(java8obj.toString());
-                if (java8 != null && java8) {
-                    typeDeclaration = "";
-                }
-            }
+            // comment out below for JDK7
```
Comment (welshm) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractJavaCodegen.java:910]: Same - can be removed?

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaJAXRSSpecServerCodegen.java [hunk 2]
```diff
@@ -127,7 +127,8 @@ public void processOpts() {
             if (!supportAsync) {
                 additionalProperties.remove(SUPPORT_ASYNC);
             } else {
-                setJava8ModeAndAdditionalProperties(true);
+                // java8 tag has been deprecated
```
Comment (welshm) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaJAXRSSpecServerCodegen.java:130]: What is the intent of keeping these comments? To remove them after showing a JDK8 upgrade is stable?

Comment (wing328) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaJAXRSSpecServerCodegen.java:130]: just leave a mark there what the `else` clause is for.

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractJavaCodegen.java (hunk 1 above)
Comment (wing328) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractJavaCodegen.java:910]: sure, will remove those in one of the upcoming PRs

Comment (wing328) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractJavaCodegen.java:910]: Done via 1892f2f9931

**Pull request merged**